backend/cn/lib/explain.ml

```diff
@@ -26,6 +26,430 @@ type log_entry =
 
 type log = log_entry list (* most recent first *)
 
+(** Infrastructure for checking if a countermodel satisfies a predicate **)
```

Comment:
Can you add a little bit more about the terminology and strategy here too? The key functions to look at, how they relate to each other.